tests/functional/src/main/java/org/mule/functional/classloading/isolation/classification/ExtensionPluginUrlClassification.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.functional.classloading.isolation.classification;
+
+import java.net.URL;
+import java.util.List;
+
+/**
+ * Represents a URL classification for an {@link org.mule.runtime.extension.api.annotation.Extension}
+ *
+ * @since 4.0
+ */
+public class ExtensionPluginUrlClassification extends PluginUrlClassification
+{
+
+    private Class extension;
+
+    public ExtensionPluginUrlClassification(Class extension, List<URL> urls)
+    {
+        super(extension.getName(), urls);
+        this.extension = extension;
+    }
+
+    public Class getExtension()
```

Comment:
You are right, I added this late Yesterday and it won't be necessary. I'll
change this and the logic to check if it is an extension
artifactClassLoader when registering extensions in the contextBuilder.

On Mon, Jul 11, 2016 at 9:46 AM, Pablo Kraan <notifications@github.com>
wrote:

> In
> tests/functional/src/main/java/org/mule/functional/classloading/isolation/classification/ExtensionPluginUrlClassification.java
> <https://github.com/mulesoft/mule/pull/2930#discussion_r70250130>:
>
> > + * Represents a URL classification for an {@link org.mule.runtime.extension.api.annotation.Extension}
> > + *
> > + * @since 4.0
> > + */
> > +public class ExtensionPluginUrlClassification extends PluginUrlClassification
> > +{
> > +
> > +    private Class extension;
> > +
> > +    public ExtensionPluginUrlClassification(Class extension, List<URL> urls)
> > +    {
> > +        super(extension.getName(), urls);
> > +        this.extension = extension;
> > +    }
> > +
> > +    public Class getExtension()
>
> Seems to me that this method is never used. And if that is correct, this
> class should be removed and the check to determine whether or not a plugin
> is a extension should be done differently (like checking if the plugin has
> an extension describer resource)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/mulesoft/mule/pull/2930/files/53339c04d7accbf6988fcbb2a79d0b00c40a8805#r70250130>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AG96KdHgXKj5pbKPko-UYzYRngOvTNONks5qUjtAgaJpZM4JGYaT>
> .
>



-- 
<http://www.mulesoft.com/>
Guillermo Fernandes, Senior Software Engineer

Av. Corrientes 316 Entre Piso, Buenos Aires, C1043AAQ, Argentina


<https://www.linkedin.com/company/www.mulesoft.com>
<https://twitter.com/mulesoft>   <https://www.facebook.com/MuleSoft>
<https://www.youtube.com/user/mulesoftvids> We're hiring!
<http://www.mulesoft.com/careers>